x/incentives/types/gov.go

```diff
@@ -0,0 +1,73 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+const (
+	ProposalTypeCreateGaugeGroups = "CreateGaugeGroups"
+)
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeCreateGaugeGroups)
+	govtypes.RegisterProposalTypeCodec(&CreateGaugeGroupsProposal{}, "osmosis/CreateGaugeGroupsProposal")
+}
+
+var (
+	_ govtypes.Content = &CreateGaugeGroupsProposal{}
+)
+
+// NewCreateGaugeGroupsProposal returns a new instance of a gauge group creation proposal struct.
+func NewCreateGaugeGroupsProposal(title, description string, groups []CreateGroup) govtypes.Content {
+	return &CreateGaugeGroupsProposal{
+		Title:        title,
+		Description:  description,
+		CreateGroups: groups,
+	}
+}
+
+func (p *CreateGaugeGroupsProposal) GetTitle() string { return p.Title }
+
+// GetDescription gets the description of the proposal
+func (p *CreateGaugeGroupsProposal) GetDescription() string { return p.Description }
+
+// ProposalRoute returns the router key for the proposal
+func (p *CreateGaugeGroupsProposal) ProposalRoute() string { return RouterKey }
+
+// ProposalType returns the type of the proposal
+func (p *CreateGaugeGroupsProposal) ProposalType() string {
+	return ProposalTypeCreateGaugeGroups
+}
+
+// ValidateBasic validates a governance proposal's abstract and basic contents
+func (p *CreateGaugeGroupsProposal) ValidateBasic() error {
```

Comment:
Please add check that at least 2 pool IDs are given. It is invalid otherwise

Reply:
That is already done on lines 51 to 55